### ISSUE TITLE
$or operator behaviour mismatch with MongoDB

### ISSUE THREAD

**Issue by nabati** [crcn/sift.js#227]
Running the following in MongoDB shell returns an error
```
db.data.find({
  responsible: { $or: [{ name: "Poyan" }, { name: "Marcus" }] }
});
```

`Unknown operator $or`.

However, the same query filters correctly with sift. 

```
sift({
  responsible: { $or: [{ name: "Poyan" }, { name: "Marcus" }] }
})({responsible: {name: "Poyan"}) // => true
```

In MongoDB instead, we have to write

```
db.data.find({
  $or: [
    { responsible: { name: Poyan } },
    { responsible: { name: Marcus }}
  ]
})

```

Is this mismatch in behaviour intentional? If so, I didn't find any mention of it in the documentation. 

As a developer, I would've expected sift to throw or otherwise return an error when trying to run this query, similar to who Mongo is doing it.

**Comment by nabati** [comment]
Any thoughts on this @crcn, on a principle level?

**Comment by crcn** [comment]
Sift behavior should be a 1:1 map with Mongo, so this is a bug. Thanks for reporting it! 

Since this a major change, I published a fix to NPM as `v14.0.0`. This also includes a fix for `$elemMatch`. 

Lmk if this fixes the issue and I'll go ahead and close this ticket.

**Comment by nabati** [comment]
This looks good, thank you!
